yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -206,6 +206,62 @@ func (d *RangeTableFuncDetector) GetIssues() []QueryIssue {
 	return issues
 }
 
+type JsonbSubscriptingDetector struct {
+	query          string
+	jsonbColumns   []string
+	detected       bool
+	jsonbFunctions []string
+}
+
+func NewJsonbSubscriptingDetector(query string, jsonbColumns []string, jsonbFunctions []string) *JsonbSubscriptingDetector {
+	return &JsonbSubscriptingDetector{
+		query:          query,
+		jsonbColumns:   jsonbColumns,
+		jsonbFunctions: jsonbFunctions,
+	}
+}
+
+func (j *JsonbSubscriptingDetector) Detect(msg protoreflect.Message) error {
+
+	if queryparser.GetMsgFullName(msg) != queryparser.PG_QUERY_A_INDIRECTION_NODE {
+		return nil
+	}
+	aIndirectionNode, ok := queryparser.GetAIndirectionNode(msg)
+	if !ok {
+		return nil
+	}
+
+	arg := aIndirectionNode.GetArg()
```

Comment:
can you add a comment here explaining the indirection node and the arg, indices fields at a high level. Explain that we're essentially checking that the arg is a jsonb type